### ISSUE TITLE
How long do i have to train?

### ISSUE THREAD

**Issue by YangChangHee** [lschmidtke/shape_templates#3]
thanks for your work!
After reading the paper, I`m very impressed and turning the training around to see if it is feasible
<Human 3.6m Dataset Training>
Frame 1 => Subject 1 <All>
Frame 2 => Subject 1 <Same Cam of Frame 1> and Fixed Image

Ex) 
Cam1 => Subject 1, action_idx 2, subaction_idx 1, cam_idx 1
Cam2 => Subject 1, action_idx 2, subaction_idx 1, cam_idx 2
Cam3 => Subject 1, action_idx 2, subaction_idx 1, cam_idx 3
Cam4 => Subject 1, action_idx 2, subaction_idx 1, cam_idx 4

Ex) Frame 1 => Subject 1, action_idx 2, subaction_idx 2, cam_idx 1
     Frame 2 => Subject 1, action_idx 2, subaction_idx 1, cam_idx 1

Frame 1
![image](https://user-images.githubusercontent.com/59610723/153337727-f82e9db4-fa1b-49f3-a2f3-ae644b51018a.png)

Frame 2
![image](https://user-images.githubusercontent.com/59610723/153337748-e09e97f8-f37f-4921-b192-0fe7f3464af4.png)

reconstructed_frame
![image](https://user-images.githubusercontent.com/59610723/153337789-4abdb1b4-cda4-4b0f-8a9e-f6d78daf7dfb.png)

transformed_template
![image](https://user-images.githubusercontent.com/59610723/153337828-486f943b-aafd-444c-9108-24066469b82d.png)

Losses
![image](https://user-images.githubusercontent.com/59610723/153337875-495a394d-1f72-4c4c-8712-248031df668b.png)

The Loss is decreasing, but it takes too long
Is this because i didn`t match the background? or
Do i need more Training?

If i must to match the background , tell me how to match it please.
Thank you

**Comment by lschmidtke** [comment]
Hi! Yes, the background of frame 1 should match that of frame 2. See [here](https://github.com/lschmidtke/shape_templates/issues/2) for details on how to do that.

**Comment by YangChangHee** [comment]
Thank you for your answer

I'd like to ask a few more questions
1. When i match the background, does the subject have to be in the same position?
case 1. (frame 1 subject => center, frame 2 subject => center)
case 2. (frame 1 subject => center, frame 2 subject => corner)

so my training scenario includes both case 1 and case 2 but your suggested method only seems to work for case 1 am i right?

2. also i would like to know how much training time was taken to obtain the results on your paper and the training specs(GPU).

**Comment by lschmidtke** [comment]
1. No, the subjects can be at different locations, as long as the background stays the same.
2. I trained for 20 epochs on a titan rtx which took around 4-5 days. Learning rate was 1e-4, but I did not really tune that. 
In fact it might make sense to go a bit higher.

Hope that helps!

**Comment by YangChangHee** [comment]
thank you for your kindness
